OpenRA.Game/Traits/LintAttributes.cs

```diff
@@ -94,4 +94,14 @@ public PaletteReferenceAttribute(string playerPaletteReferenceSwitch)
 			PlayerPaletteReferenceSwitch = playerPaletteReferenceSwitch;
 		}
 	}
+
+	[AttributeUsage(AttributeTargets.Class)]
+	public sealed class TraitLocationAttribute : Attribute
+	{
+		public readonly SystemActors[] SystemActors;
```

Comment:
Switching to enums also removed the need to use an array here. Define `SystemActors` with `[Flags]` and give the members power-of-two values, then values can be composed using normal bitwise operations. Search for `[Flags]` and follow references to find examples of the many other places where we do this.

Reply:
Redid it with flagged enums.